atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/AbstractDbWriteTable.java

```diff
@@ -133,11 +135,25 @@ public void delete(List<Entry<Cell, Long>> entries) {
 
         String prefixedTableName = prefixedTableNames.get(tableRef);
         conns.get().updateManyUnregisteredQuery(" /* DELETE_ONE (" + prefixedTableName + ") */ "
-                + " DELETE /*+ INDEX(m pk_" + prefixedTableName + ") */ "
+                + " DELETE /*+ INDEX(m " + getPrimaryKeyConstraintName(prefixedTableName) + ") */ "
                 + " FROM " + prefixedTableName + " m "
                 + " WHERE m.row_name = ? "
                 + "  AND m.col_name = ? "
                 + "  AND m.ts = ?",
                 args);
     }
+
+    private String getPrimaryKeyConstraintName(String tableName) {
+        String pkConstraintName = "pk_" + tableName;
+        if (config.type().equals(OracleDdlConfig.TYPE)) {
+            return truncateToMaxOracleLength(pkConstraintName);
+        }
+        return pkConstraintName;
+    }
+
+    private String truncateToMaxOracleLength(String constraintName) {
+        return constraintName
+                .substring(0, Math.min(OracleTableNameMapper.ORACLE_MAX_TABLE_NAME_LENGTH, constraintName.length()));
```

Comment:
@schlosna this is the primary key constraint name, what kind of errors do we want to throw here?